### ISSUE TITLE
Websockets over SSL crashes due to relative/absolute pathing issue

### ISSUE THREAD

**Issue by sudosammy** [beefproject/beef#1464]
#### Environment

Version: `0.4.7.0-alpha` pulled `05 Nov 2017`

Ruby: `ruby 2.3.1p112 (2016-04-26) [x86_64-linux-gnu]`

Broswer: `Chrome 61`

Server OS: `Ubuntu 16.04.3 LTS`
Victim OS: `Win 10`

#### Configuration

Config changed to support HTTPS and secure websockets:

```
        websocket:
            enable: true
            port: 61985 # WS: good success rate through proxies
            # Use encrypted 'WebSocketSecure'
            # NOTE: works only on HTTPS domains and with HTTPS support enabled in BeEF
            secure: true
            secure_port: 61986 # WSSecure
            ws_poll_timeout: 1000 # poll BeEF every second
            ws_connect_timeout: 500 # useful to help fingerprinting finish before establishing the WS channel

        # Imitate a specified web server (default root page, 404 default error page, 'Server' HTTP response header)
        web_server_imitation:
            enable: true
            type: "apache" # Supported: apache, iis, nginx
            hook_404: false # inject BeEF hook in HTTP 404 responses
            hook_root: false # inject BeEF hook in the server home page
        # Experimental HTTPS support for the hook / admin / all other Thin managed web services
        https:
            enable: true
            # In production environments, be sure to use a valid certificate signed for the value
            # used in beef.http.public (the domain name of the server where you run BeEF)
            key: "/etc/letsencrypt/live/XXX.XXX/privkey.pem"
            cert: "/etc/letsencrypt/live/XXX.XXX/fullchain.pem"
```

Have you enabled or disabled any BeEF extensions?
No

#### Summary

When attempting to hook with secure websockets, BeEF crashes as it can't find the certificate files. This is because it searches for them relative to the current working directory and not at the absolute path specified in the `config.yaml` file.

#### Expected Behaviour

BeEF locates the certificate files at the absolute path specified in the config file: `/etc/letsencrypt/live/XXX.XXX/` and does not crash.

#### Actual Behaviour

BeEF crashes with the following error when it can't find the certificate files:

```
/var/lib/gems/2.3.0/gems/eventmachine-1.0.9.1/lib/em/connection.rb:411:in `block in start_tls': Could not find /home/username/beef//etc/letsencrypt/live/XXX.XXX/privkey.pem for start_tls (EventMachine::FileNotFoundException)
        from /var/lib/gems/2.3.0/gems/eventmachine-1.0.9.1/lib/em/connection.rb:409:in `each'
        from /var/lib/gems/2.3.0/gems/eventmachine-1.0.9.1/lib/em/connection.rb:409:in `start_tls'
        from /var/lib/gems/2.3.0/gems/em-websocket-0.5.1/lib/em-websocket/connection.rb:69:in `post_init'
        from /var/lib/gems/2.3.0/gems/eventmachine-1.0.9.1/lib/em/connection.rb:58:in `block in new'
        from /var/lib/gems/2.3.0/gems/eventmachine-1.0.9.1/lib/em/connection.rb:49:in `instance_eval'
        from /var/lib/gems/2.3.0/gems/eventmachine-1.0.9.1/lib/em/connection.rb:49:in `new'
        from /var/lib/gems/2.3.0/gems/eventmachine-1.0.9.1/lib/eventmachine.rb:1497:in `event_callback'
        from /var/lib/gems/2.3.0/gems/eventmachine-1.0.9.1/lib/eventmachine.rb:193:in `run_machine'
        from /var/lib/gems/2.3.0/gems/eventmachine-1.0.9.1/lib/eventmachine.rb:193:in `run'
        from /var/lib/gems/2.3.0/gems/thin-1.7.2/lib/thin/backends/base.rb:73:in `start'
        from /var/lib/gems/2.3.0/gems/thin-1.7.2/lib/thin/server.rb:162:in `start'
        from /home/username/beef/core/main/server.rb:125:in `start'
        from beef:182:in `<main>'
```

Workaround: Satify the exception by copying the files to `/home/username/beef//etc/letsencrypt/live/XXX.XXX/`

#### Steps to Reproduce

Configure BeEF to use an SSL certificate at an absolute path in `config.yaml` and also enable websockets support. Attempt to hook a page and observe the crash.

**Comment by H4xl0r** [comment]
Easy to fix , take the cert files and move them to beef home directory and then put just the names into the config.yaml ..... 
then there is no needed path

**Comment by bcoles** [comment]
Hey samy. This should now be resolved; as of commit 8e90587a774b2de9ad1c0785762e49c6c0abe616.

Unfortunately there appears to be some other issues with websocket comms.
